### PR TITLE
SorrTask694_unit_test_get_service_resource

### PR DESCRIPTION
**Pull request by tkpratardan**
Task #694 
@samarth9008 @sonaalKant 

Created a unit test to check the s3 service. 

Review request:
1. Am I headed in the right direction here?
2. Should I add more unit tests to check other services? If yes, could you mention what all services?

cc : @gpsaggese

### CODE REVIEW

File: helpers/test/test_haws.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import io
+import os
+
+import helpers.haws as haws
+import helpers.hunit_test as hunitest
+
+_LOG = logging.getLogger(__name__)
+
+
+class test_get_service_resource(hunitest.TestCase):
```
Comment (sonaalKant) [inline helpers/test/test_haws.py:14]: -> `Test_get_service_resource`
refer https://github.com/sorrentum/sorrentum/blob/e841aaa34c71b200f5405e7257cc108a57086fd8/docs/coding/all.write_unit_tests.how_to_guide.md#conventions

File: helpers/test/test_haws.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    @mock_s3
+    def test_s3_resource(self):
+        """
+        Test that get_service_resource correctly retrieves an S3 resource.
+        """
+        aws_profile = "__mock__"  
+        service_name = "s3"
+
+       #Creating a mock s3 bucket.
```
Comment (sonaalKant) [inline helpers/test/test_haws.py:35]: Nits. Indent correctly.

File: helpers/test/test_haws.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        Test that get_service_resource correctly retrieves an S3 resource.
+        """
+        aws_profile = "__mock__"  
+        service_name = "s3"
+
+       #Creating a mock s3 bucket. 
+        s3 = boto3.resource('s3')
+        s3.create_bucket(Bucket='my-test-bucket')
+
+
```
Comment (sonaalKant) [inline helpers/test/test_haws.py:39]: Remove blank spaces. Always run linter before pushing code, `i lint -f "fillename"`

File: helpers/test/test_haws.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        Mocked AWS Credentials for moto.
+        """
+        os.environ["MOCK_AWS_ACCESS_KEY_ID"] = "testing"
+        os.environ["MOCK_AWS_SECRET_ACCESS_KEY"] = "testing"
+        os.environ["MOCK_AWS_SECURITY_TOKEN"] = "testing"
+        os.environ["MOCK_AWS_SESSION_TOKEN"] = "testing"
+        os.environ["MOCK_AWS_DEFAULT_REGION"] = "us-east-1"
+
+    @mock_s3
+    def test_s3_resource(self):
```
Comment (sonaalKant) [inline helpers/test/test_haws.py:28]: Nits
-> `test1(self) -> None`

File: helpers/test/test_haws.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import helpers.haws as haws
+import helpers.hunit_test as hunitest
+
+_LOG = logging.getLogger(__name__)
+
+
+class test_get_service_resource(hunitest.TestCase):
+
+    @pytest.fixture(autouse=True, scope="class")
+    def aws_credentials(self):
```
Comment (sonaalKant) [inline helpers/test/test_haws.py:17]: Nits
-> `aws_credentials(self) -> None`

**Review by sonaalKant — CHANGES_REQUESTED** [review]
Overall good job. Few things to take care for future:
- Run linter. (It will warn you with most of the nits I mentioned)
- Refer documentations for conventions that we follow.

**Comment by tkpratardan** [comment]
@sonaalKant 

Thank you for your review. I apologize for the issues. Will make sure to keep my code consistent with the conventions and formatting further on.

for now, I have made the required changes, and made the commit after running the file with linter. Could you please review it? Also please let me know if I have to add any more tests.

File: helpers/test/test_haws.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        os.environ["MOCK_AWS_ACCESS_KEY_ID"] = "testing"
+        os.environ["MOCK_AWS_SECRET_ACCESS_KEY"] = "testing"
+        os.environ["MOCK_AWS_SECURITY_TOKEN"] = "testing"
+        os.environ["MOCK_AWS_SESSION_TOKEN"] = "testing"
+        os.environ["MOCK_AWS_DEFAULT_REGION"] = "us-east-1"
+
+    @mock_s3
+    def test1(self) -> None:
+        """
+        Test that haws.get_service_resource() correctly retrieves an S3
```
Comment (samarth9008) [inline helpers/test/test_haws.py:29]: haws.get_service_resource() -> `haws.get_service_resource()`. 

Surrounded by backticks.

Also no `an` before s3

File: helpers/test/test_haws.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        resource.
+        """
+        aws_profile = "__mock__"
+        service_name = "s3"
+        # Creating a mock s3 bucket.
```
Comment (samarth9008) [inline helpers/test/test_haws.py:34]: -> # Create mock `s3` bucket.

File: helpers/test/test_haws.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        s3 = boto3.resource("s3")
+        s3.create_bucket(Bucket="my-test-bucket")
+        s3_resource = haws.get_service_resource(aws_profile, service_name)
+        buckets = list(s3_resource.buckets.all())
```
Comment (samarth9008) [inline helpers/test/test_haws.py:38]: Add comment above line 38

```
# Get all `s3` buckets.
```

File: helpers/test/test_haws.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        bucket_names = [bucket.name for bucket in buckets]
+        # Assert
+        self.assertIn("my-test-bucket", bucket_names)
```
Comment (samarth9008) [inline helpers/test/test_haws.py:41]: Lets use our internal function for asserting

```
import helpers.hdbg as hdbg #import at the top
hdbg.dassert_in("my-test-bucket", bucket_names)
```

File: helpers/test/test_haws.py [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        """
+        aws_profile = "__mock__"
+        service_name = "s3"
+        # Creating a mock s3 bucket.
+        s3 = boto3.resource("s3")
+        s3.create_bucket(Bucket="my-test-bucket")
+        s3_resource = haws.get_service_resource(aws_profile, service_name)
+        buckets = list(s3_resource.buckets.all())
+        bucket_names = [bucket.name for bucket in buckets]
+        # Assert
```
Comment (samarth9008) [inline helpers/test/test_haws.py:40]: -> `# Check.`

File: helpers/test/test_haws.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+import helpers.hunit_test as hunitest
+
+_LOG = logging.getLogger(__name__)
+
+
+class Test_get_service_resource(hunitest.TestCase):
+    @pytest.fixture(autouse=True, scope="class")
+    def aws_credentials(self) -> None:
+        """
+        Mocked AWS Credentials for moto.
```
Comment (samarth9008) [inline helpers/test/test_haws.py:18]: Credentials  -> credentials

**Review by samarth9008 — CHANGES_REQUESTED** [review]
Good improvements. Few changes from my side.

**Comment by samarth9008** [comment]
Pls resolve the conversation if they are addressed

**Comment by tkpratardan** [comment]
@samarth9008 
Thank you for pointing out the inconsistencies. I have made the changes, please verify.

**Comment by sonaalKant** [comment]
LGTM.

**Review by sonaalKant — APPROVED** [review]

File: helpers/test/test_haws.py [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+import boto3
+import pytest
+from moto import mock_s3
+
+import helpers.haws as haws
+import helpers.hdbg as hdbg
+import helpers.hunit_test as hunitest
+
+_LOG = logging.getLogger(__name__)
```
Comment (samarth9008) [inline helpers/test/test_haws.py:12]: Do we need it?

**Review by samarth9008 — COMMENTED** [review]
A question and LGTM

File: helpers/test/test_haws.py (hunk 12 above)
Comment (tkpratardan) [reply helpers/test/test_haws.py:12]: Actually, no. I just followed the helpers/unit_test_skeleton.py. Should I remove it?

Comment (samarth9008) [reply helpers/test/test_haws.py:12]: If `_LOG` not used anywhere in the code and then lets remove it.

Comment (tkpratardan) [reply helpers/test/test_haws.py]: Done, please check and verify.

**Review by gpsaggese — COMMENTED** [review]
LGTM

**Pull request merged**